src/matrix_cookbook/mat_inv_lib.lean

```diff
@@ -0,0 +1,197 @@
+import linear_algebra.matrix.nonsingular_inverse
+import linear_algebra.matrix.pos_def
+import linear_algebra.matrix.spectrum
+import data.complex.basic
+
+variables {m n p : Type*}
+variables [fintype m] [fintype n] [fintype p]
+variables [decidable_eq m] [decidable_eq n] [decidable_eq p]
+
+open matrix
+open_locale matrix big_operators
+
+
+lemma is_unit_of_pos_def {A : matrix m m ℂ} (hA: matrix.pos_def A): 
+  is_unit A.det := 
+begin
+  rw is_unit_iff_ne_zero,
+  apply ne.symm,
+  cases hA with hAH hpos,
+  rw is_hermitian.det_eq_prod_eigenvalues hAH,
+  norm_cast,
+  rw ← ne.def,
+  apply ne_of_lt,
+  apply finset.prod_pos,
+  intros i _,
+  rw hAH.eigenvalues_eq,
+  apply hpos _ (λ h, _),
+  have h_det : (hAH.eigenvector_matrix)ᵀ.det = 0,
+    from matrix.det_eq_zero_of_row_eq_zero i (λ j, congr_fun h j),
+  simpa only [h_det, not_is_unit_zero] using
+    is_unit_det_of_invertible hAH.eigenvector_matrixᵀ,
+end
```

Comment:
This follows trivially from [`matrix.pos_def.det_pos`](https://leanprover-community.github.io/mathlib_docs/linear_algebra/matrix/pos_def.html#matrix.pos_def.det_pos) I think; so there's no real point having it.

Reply:
It seems this one is a problem. We know because the matrix is positive definite that its determinant is a real number. But the way det is defined means for a $\mathbb{C}$ matrix the determinant is another complex number. Complex numbers have no less than operator as far as I know.

I think that is why the person who wrote `matrix.pos_def.det_pos` avoided it anyway. just to avoid the headache?

So I guess the question is given a matrix that is `is_R_or_C` how do you tell lean hey lean I know this determinant is going to be real number deal with it as such?